ocaml/typing/typedtree.ml

```diff
@@ -217,11 +217,15 @@ and comprehension_iterator =
       ; sequence : expression }
 
 and 'k case =
-    {
-     c_lhs: 'k general_pattern;
-     c_guard: expression option;
-     c_rhs: expression;
-    }
+  {
+    c_lhs: 'k general_pattern;
+    c_guard: guard option;
+    c_rhs: expression;
+  }
+  
+and guard =
+  | Predicate of expression
```

Comment:
As discussed, let's call these "Guard_predicate" and "Guard_pattern" instead (and maybe propagate the name to typecore). I'm fine CR-sooning and doing in another PR.

Reply:
CR-sooned